### PR TITLE
Fix cage selection bugs and ignore caged vermin

### PR DESCRIPTION
**Pull request by arekatir**

### CODE REVIEW

File: stripcaged.lua [hunk 1]
```diff
@@ -1,8 +1,29 @@
+local rawargs = {...}
+local args = require 'utils'.invert(rawargs)
```
Comment (myk002) [inline stripcaged.lua:2]: you are better off using `argparse.processArgsGetopt()`. it returns the non-option params as a list.

File: stripcaged.lua [hunk 2]
```diff
@@ -1,8 +1,29 @@
...
+
 local function plural(nr, name)
     -- '1 cage' / '4 cages'
     return string.format('%s %s%s', nr, name, nr ~= 1 and 's' or '')
 end
 
+local function isexcludedvermin(item)
```
Comment (myk002) [inline stripcaged.lua:9]: consider passing an `opts` table here instead of reading from global args lists

File: stripcaged.lua (hunk 1 above)
Comment (arekatir) [reply stripcaged.lua]: Thanks. My latest commits try to address this.

File: stripcaged.lua [hunk 3]
```diff
@@ -252,8 +263,13 @@ else
     end
 end
 
+-- Print help if requested
+if opts.help then
+    print(dfhack.script_help())
```
Comment (myk002) [inline stripcaged.lua:268]:
```suggestion
    print(dfhack.script_help())
    return
```

**Review by myk002 — APPROVED** [review]

**Pull request merged**